warzone/Orders.h

```diff
@@ -14,112 +18,130 @@ class Order {
-	DeployOrder* operator= (const DeployOrder& o);
+	int numArmies;
+	Territory* target;
+
+	friend std::ostream& operator<<(std::ostream& strm, const DeployOrder& o);
+	DeployOrder& operator= (const DeployOrder& o);
 };
 
 
 class AdvanceOrder : public Order {
 public:
-	AdvanceOrder(Player* player);
+	AdvanceOrder(Player* player, int numArmies, Territory* source, Territory* target);
 	AdvanceOrder(AdvanceOrder* other);
 	~AdvanceOrder();
 
 	bool validate();
 	bool execute();
-	string toString() const;
+	void battle();
+	std::string toString() const;
 
 private:
-	friend ostream& operator<<(ostream& strm, const AdvanceOrder& o);
-	AdvanceOrder* operator= (const AdvanceOrder& o);
+	int numArmies;
+	Territory* source;
+	Territory* target;
+
+	friend std::ostream& operator<<(std::ostream& strm, const AdvanceOrder& o);
+	AdvanceOrder& operator= (const AdvanceOrder& o);
 };
 
 
 class BombOrder : public Order {
 public:
-	BombOrder(Player* player);
+	BombOrder(Player* player, Territory* target);
 	BombOrder(BombOrder* other);
 	~BombOrder();
 
 	bool validate();
 	bool execute();
-	string toString() const;
+	std::string toString() const;
 
 private:
-	friend ostream& operator<<(ostream& strm, const BombOrder& o);
-	BombOrder* operator= (const BombOrder& o);
+	Territory* target;
+
+	friend std::ostream& operator<<(std::ostream& strm, const BombOrder& o);
+	BombOrder& operator= (const BombOrder& o);
 };
 
 
 class BlockadeOrder : public Order {
 public:
-	BlockadeOrder(Player* player);
+	BlockadeOrder(Player* player, Territory* target);
 	BlockadeOrder(BlockadeOrder* other);
 	~BlockadeOrder();
 
 	bool validate();
 	bool execute();
-	string toString() const;
+	std::string toString() const;
 
 private:
-	friend ostream& operator<<(ostream& strm, const BlockadeOrder& o);
-	BlockadeOrder* operator= (const BlockadeOrder& o);
+	Territory* target;
+
+	friend std::ostream& operator<<(std::ostream& strm, const BlockadeOrder& o);
+	BlockadeOrder& operator= (const BlockadeOrder& o);
 };
 
 
 class AirliftOrder : public Order {
 public:
-	AirliftOrder(Player* player);
+	AirliftOrder(Player* player, int numArmies, Territory* source, Territory* target);
 	AirliftOrder(AirliftOrder* other);
 	~AirliftOrder();
 
 	bool validate();
 	bool execute();
-	string toString() const;
+	std::string toString() const;
 
 private:
-	friend ostream& operator<<(ostream& strm, const AirliftOrder& o);
-	AirliftOrder* operator= (const AirliftOrder& o);
+	int numArmies;
+	Territory* source;
+	Territory* target;
+
+	friend std::ostream& operator<<(std::ostream& strm, const AirliftOrder& o);
+	AirliftOrder& operator= (const AirliftOrder& o);
 };
 
 
 class NegotiateOrder : public Order {
 public:
-	NegotiateOrder(Player* player);
+	NegotiateOrder(Player* player, Player* targeted);
 	NegotiateOrder(NegotiateOrder* other);
 	~NegotiateOrder();
 
 	bool validate();
 	bool execute();
-	string toString() const;
+	std::string toString() const;
```

Comment:
Why does this have a toString? Shouldn't we just use <<?

Reply:
Something about it being a subclass so it wasn't working? @felixlapierre helped me fix it but I'm not remembering the reason. It was like this for A1 also.